package cetty.zurg.slave;

option java_package = "io.cetty.zurg.slave";
option java_outer_classname = "SlaveProto";

message DiskUsage {
  required string device = 1;
  required string  mount = 2;

  // statvfs
  required int32   block_size = 3;
  required int64   blocks_all = 4;
  required int64  blocks_free = 5;
  required int64 blocks_avail = 6;

  required int64   inodes_all = 7;
  required int64  inodes_free = 8;
  required int64 inodes_avail = 9;

  required uint64    fsid = 10;
  required uint64    flag = 11;
  required uint64 namemax = 12;
}

// ------------- Application

enum ApplicationState {
  kUnknown = 0;
  kNewApp  = 1;
  kRunning = 2;
  kExited  = 3;
  kError   = 4;
}

message ApplicationStatus {
  required ApplicationState state = 1 [default = kUnknown];
  required string            name = 2;
  optional int32              pid = 3;
  optional string executable_file = 4;
  optional string             cwd = 5;
  optional string         message = 6;
  optional string      slave_name = 8;

  optional int64     start_time_us = 10;
  optional int64 last_stop_time_us = 11;
  optional string        proc_stat = 12;
  optional string      proc_status = 13;

  optional int32  last_exit_status = 30;
  optional int32     last_signaled = 31;
  optional bool      last_coredump = 32;
}
